companydata/gap-sogessur-emit.go

```diff
@@ -0,0 +1,474 @@
+		"NAZIONE PROPRIETARIO",
+		"CODICE FISCALE PROPRIETARIO",
+		"PARTITA IVA PROPRIETARIO",
+		"PR_SESSO",
+		"PR_DATA_NASC",
+		"PR_LUOGO_NASC",
+		"PR_PROV_NASC",
+	}
+}
+
+func getGapRowMap(policy models.Policy, transaction models.Transaction) map[string]string {
+	vehicle := policy.Assets[0].Vehicle
+	contractor := policy.Contractor
+	policyHolder := policy.Assets[0].Person
+	vehicleOwner := policy.Contractor // NOTE: For the moment assuming vehicleOwner is the contractor
+	offerName := policy.OfferlName
+
+	genders := []string{"F", "M", "G"} // For validation
+	// Maps for replacing the values in policy into valid values for the gap csv
+	vehicleTypes := map[string]string{
+		"Autocarro":   "C",
+		"Autoveicolo": "A",
+		"Motociclo":   "M",
+		"Camper":      "P",
+	}
+	boolAnswers := map[bool]string{
+		true:  "SI",
+		false: "NO",
+	}
+	gapOptions := map[string]string{
+		"base":     "OPTION 1",
+		"complete": "OPTION 2",
+	}
+	isVehicleNew := map[string]string{
+		"Nuovo": "SI",
+		"Usato": "NO",
+	}
+	powerSupplyCodes := map[string]string{
+		"Sì": "E",
+		"No": "",
+	}
+	// Assuming we have only this payment type
+	offer := policy.OffersPrices[offerName][string(models.PaySingleInstallment)]
+	vehicleWeight := ""
+	if vehicle.Weight > 0 {
+		vehicleWeight = strconv.Itoa(int(vehicle.Weight))
+	}
+	zoneCode := quote.GetAreaByProvince(policyHolder.Residence.CityCode)
+	return map[string]string{
+		"NUMERO POLIZZA":                       CheckIfIsAlphaNumeric(policy.CodeCompany),
+		"NUMERO CONTRATTO":                     CheckIfIsAlphaNumeric(policy.CodeCompany),
+		"TIPO OPERAZIONE":                      "A", // 'A' = Subscription
+		"DATA OPERAZIONE":                      policy.StartDate.Format(gapDateFormat),
+		"COGNOME/RAGIONE SOCIALE ASSICURATO":   policyHolder.Surname,
+		"NOME ASSICURATO":                      policyHolder.Name,
+		"INDIRIZZO ASSICURATO":                 getAddress(*policyHolder.Residence),
+		"COMUNE ASSICURATO":                    policyHolder.Residence.Locality,
+		"CAP ASSICURATO":                       CheckIfIsNumeric(policyHolder.Residence.PostalCode),
+		"PROVINCIA ASSICURATO":                 CheckIfIsAlphaNumeric(policyHolder.Residence.CityCode),
+		"NAZIONE ASSICURATO":                   "Italia",
+		"CODICE FISCALE ASSICURATO":            CheckIfIsAlphaNumeric(policyHolder.FiscalCode),
+		"PARTITA IVA ASSICURATO":               CheckIfIsNumeric(policyHolder.VatCode),
+		"TARGA":                                CheckIfIsAlphaNumeric(vehicle.Plate),
+		"TELAIO":                               "",
+		"MODELLO":                              vehicle.Model,
+		"MARCA":                                vehicle.Manufacturer,
+		"CILINDRATA":                           "",
+		"ALIMENTAZIONE":                        powerSupplyCodes[vehicle.PowerSupply],
+		"ANTIFURTO SATELLITARE":                boolAnswers[vehicle.HasSatellite],
+		"TIPO VEICOLO":                         vehicleTypes[vehicle.VehicleType],
```

Comment:
use VehicleTypeDesc or change mapping to use VehicleType values (e.g., car, truck, camper)